lib/api/core/plugins/pluginsManager.js

```diff
@@ -767,20 +801,228 @@ function registerHook(kuzzle, plugin, event, fn) {
+        valid = false;
+      }
+
+      if (!isObject(strategy.config.strategyOptions)) {
+        // eslint-disable-next-line no-console
+        console.error(`${errorPrefix} The "strategyOptions" of the strategy "${strategyName}" must be an object.`);
+        valid = false;
+      }
+
+      if (!isObject(strategy.config.authenticateOptions)) {
+        // eslint-disable-next-line no-console
+        console.error(`${errorPrefix} The "authenticateOptions" of the strategy "${strategyName}" must be an object.`);
+        valid = false;
+      }
+
+      if (!strategy.config.fields || !Array.isArray(strategy.config.fields)) {
+        // eslint-disable-next-line no-console
+        console.error(`${errorPrefix} The "fields" of the strategy "${strategyName}" must be an array.`);
+        valid = false;
+      }
+    }
+  });
+
+  if (valid) {
+    Object.keys(plugin.strategies).forEach(strategyName => {
+      const
+        strategy = plugin.strategies[strategyName],
+        methods = {};
+
+      Object.keys(strategy.methods).filter(name => name !== 'verify').forEach(methodName => {
+        methods[methodName] = (...args) => {
+          return new Bluebird((resolve, reject) => {
+            try {
+              resolve(plugin[strategy.methods[methodName]].bind(plugin)(...args));
+            }
+            catch (error) {
+              reject(error);
+            }
+          })
+          .catch(error => Bluebird.reject(error instanceof KuzzleError ? error : new PluginImplementationError(error)));
+        };
+      });
+
+      authentications[strategyName] = {
+        strategy,
+        methods
+      };
+
+      registerStrategy(kuzzle, plugin, authentications, strategyName);
+      kuzzle.passport.injectAuthenticateOptions(strategyName, strategy.config.authenticateOptions || {});
+    });
+  }
+}
 
 /**
- * Injects a plugin's declared scopes into the Kuzzle passport wrapper
+ * Registers an authentication strategy
  *
- * @param {object} scope to inject, in the following format: {strategy: [scope, fields}, ...}
- * @param {object} passport wrapper object
+ * @param {Kuzzle} kuzzle
+ * @param {object} plugin
+ * @param {object<string, object>} authentications
+ * @param {string} name - strategy name
  */
-function injectScope(scope, passport) {
-  Object.keys(scope).forEach(strategy => {
-    // enforcing scope format
-    if (typeof strategy === 'string' && Array.isArray(scope[strategy])) {
-      passport.injectScope(strategy, scope[strategy]);
+function registerStrategy(kuzzle, plugin, authentications, name) {
+  const
+    {
+      strategy,
+      methods
+    } = authentications[name],
+    opts = Object.assign(strategy.config.strategyOptions || {}, {passReqToCallback: true}),
+    verifyAdapter = (...args) => {
+      const callback = args[args.length - 1];
+      return plugin[strategy.methods.verify](...args.slice(0, -1))
+        .then(result => {
+          if (typeof result === 'string') {
```

Comment:
that would mean to change the plugin's implementation also of course.
Note that this remark is not blocking for me.